payloads/v9/channel.ts

```diff
@@ -1207,7 +1217,35 @@ export interface APISelectMenuOption {
 	default?: boolean;
 }
 
+// TODO: Add api links
+export interface APIInputTextComponent extends APIBaseMessageComponent<ComponentType.InputText> {
+	/**
+	 * One of input text styles
+	 */
+	style: InputTextStyle;
+	/**
+	 * Text that appears on top of the input text field, max 80 characters
+	 */
+	label: string;
+	/**
+	 * Placeholder for the text input
+	 */
+	placeholder?: string;
+	/**
+	 * Minimal length of text input
+	 */
+	min_length?: number;
+	/**
+	 * Maximal length of text input
+	 */
+	max_length?: number;
+}
+
 /**
  * https://discord.com/developers/docs/interactions/message-components#message-components
  */
-export type APIMessageComponent = APIActionRowComponent | APIButtonComponent | APISelectMenuComponent;
+export type APIMessageComponent =
+	| APIActionRowComponent
+	| APIButtonComponent
+	| APISelectMenuComponent
+	| APIInputTextComponent;
```

Comment:
Well you can't use this in Messages so its not a MessageComponent :D
Probably time to define a second ActionRow for modals? Idk..

Reply:
Most likely